include/swift/AST/DiagnosticsSema.def

```diff
@@ -2790,16 +2790,17 @@ ERROR(differentiable_attr_protocol_req_where_clause,none,
 ERROR(differentiable_attr_protocol_req_assoc_func,none,
       "'@differentiable' attribute on protocol requirement cannot specify "
       "'jvp:' or 'vjp:'", ())
+ERROR(differentiable_attr_stored_property_variable_unsupported,none,
+      "'@differentiable' attribute on stored property cannot specify "
+      "'jvp:' or 'vjp:'", ())
 ERROR(differentiable_attr_empty_where_clause,none,
       "empty 'where' clause in '@differentiable' attribute", ())
 ERROR(differentiable_attr_nongeneric_trailing_where,none,
       "trailing 'where' clause in '@differentiable' attribute of non-generic function %0", (DeclName))
 ERROR(differentiable_attr_unsupported_req_kind,none,
-      "layout requirement are not supported by '@differentiable' attribute", ())
+      "'@differentiable' attribute does not support layout requirements", ())
```

Comment:
We should not have mentioned "layout requirement" anywhere in diagnostics. It is an implementation detail. Instead, we should say something like "'AnyObject' constraint".

Reply:
I just verified that `@differentiable(where T : AnyObject)` is actually visited as a conformance requirement, not a layout one. Which is inconsistent with [Generics.rst](https://github.com/apple/swift/blob/tensorflow/docs/Generics.rst) and a previous comment by @slavapestov.

I think the "class layout" requirement is currently spelled as [`T : _Class`](https://github.com/apple/swift/blob/c1dfcde1a4ddc3a6aad97571aeb38fab5f9ce37d/include/swift/AST/KnownIdentifiers.def#L165).

---

Given this, I think `'@differentiable' attribute does not support layout requirements` is a more appropriate message. Updated tests in https://github.com/apple/swift/pull/26179/commits/7ea251f981a196790baffe6031fa804dbabc8aaa. Let me know what you think.